File: 1112-find-words-that-can-be-formed-by-characters/find-words-that-can-be-formed-by-characters.cpp
class Solution {
public:
    int countCharacters(vector<string>& words, string chars) {
        
        vector<int>ch(26,0);

        for(int i=0;i<chars.size();i++){
            ch[chars[i]-'a']++;
        }

        int ans=0;

        for(int i=0;i<words.size();i++){
            string str=words[i];
            bool flag=false;
            vector<int>vec(26,0);
            for(int j=0;j<str.size();j++){
                vec[str[j]-'a']++;
            }
            for(int k=0;k<26;k++){
                if(vec[k]>ch[k]){
                    flag=true;
                    break;
                }
            }
            if(!flag){
                ans+=str.size();
            }
        }

        return ans;
    }
};

File: 1112-find-words-that-can-be-formed-by-characters/find-words-that-can-be-formed-by-characters.java
class Solution {
    public int countCharacters(String[] words, String chars) {
                
                int[] ch=new int[26];
                Arrays.fill(ch,0);

        for(int i=0;i<chars.length();i++){
            ch[chars.charAt(i)-'a']++;
        }

        int ans=0;

        for(int i=0;i<words.length;i++){
            String str=words[i];
            boolean flag=false;
            int[] vec=new int[26];
            for(int j=0;j<str.length();j++){
                vec[str.charAt(j)-'a']++;
            }
            for(int k=0;k<26;k++){
                if(vec[k]>ch[k]){
                    flag=true;
                    break;
                }
            }
            if(!flag){
                ans+=str.length();
            }
        }

        return ans;
    }
}